src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -600,9 +600,7 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 				});
 			}
 		}).finally(() => {
-			const commonKey = task.getCommonTaskId();
-			liveDependencies.delete(commonKey);
-			encounteredTasks.delete(commonKey);
+			liveDependencies.delete(task.getCommonTaskId());
```

Comment:
This still seems wrong. liveDependencies is (long) dead here; and its lifetime is being artificially extended, *just* to delete an entry from it. It also confuses the code; anyone reading this in future will assume the delete is there for a reason.

Maybe a better fix would be to introduce a new variable holding the modified dependencies... I can put up a diff...